code/modules/mob/new_player/new_player.dm

```diff
@@ -544,12 +544,17 @@
 	check_prefs_are_sane()
 	var/mob/living/carbon/human/new_character = new(loc)
 	new_character.lastarea = get_area(loc)
-
-	if(SSticker.random_players)
-		client.prefs.active_character.randomise()
-		client.prefs.active_character.real_name = random_name(client.prefs.active_character.gender)
 	client.prefs.active_character.copy_to(new_character)
-
+	if(SSticker.random_players)
+		var/mob/living/carbon/human/H = new_character
+		scramble(1, H, 100)
+		H.real_name = random_name(H.gender, H.dna.species.name) //Give them a name that makes sense for their species.
+		H.sync_organ_dna(assimilate = 1)
+		H.update_body()
+		H.reset_hair() //No more winding up with hairstyles you're not supposed to have, and blowing your cover.
```

Comment:
I take it some of these comments are from the DNA scrambler? Could do without them

Reply:
can do